File: app/src/main/java/com/andrewvychev/railwaytickets/di/components/MainComponent.java
package com.andrewvychev.railwaytickets.di.components;

import com.andrewvychev.railwaytickets.RailwayApplication;
import com.andrewvychev.railwaytickets.di.modules.MainModule;
import com.andrewvychev.railwaytickets.di.modules.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {MainModule.class, NetworkModule.class})
@Singleton
public interface MainComponent {

    void inject(RailwayApplication application);

    ScreenSubcomponent.Builder activityComponent();
}

File: app/src/main/java/com/andrewvychev/railwaytickets/di/components/MainComponent.kt
package com.andrewvychev.railwaytickets.di.components

import com.andrewvychev.railwaytickets.RailwayApplication
import com.andrewvychev.railwaytickets.di.modules.MainModule
import com.andrewvychev.railwaytickets.di.modules.NetworkModule
import dagger.Component
import javax.inject.Singleton

@Component(modules = arrayOf(MainModule::class, NetworkModule::class))
@Singleton
interface MainComponent {

    fun inject(application: RailwayApplication)

    fun activityComponent(): ScreenSubcomponent.Builder
}

